mapie/tests/test_control_risk.py

```diff
@@ -0,0 +1,194 @@
+    with pytest.raises(ValueError, match=r".*could not be broadcast*"):
+        _compute_recall(lambdas, y_preds_proba, y_toy[:-1])
+
+
+def test_compute_precision_with_wrong_shape() -> None:
+    """Test shape when using _compute_precision"""
+    with pytest.raises(ValueError, match=r".*y_pred_proba should be a 3d*"):
+        _compute_precision(lambdas, y_preds_proba.squeeze(), y_toy)
+    with pytest.raises(ValueError, match=r".*y should be a 2d*"):
+        _compute_precision(lambdas, y_preds_proba, np.expand_dims(y_toy, 2))
+    with pytest.raises(ValueError, match=r".*could not be broadcast*"):
+        _compute_precision(lambdas, y_preds_proba, y_toy[:-1])
+
+
+@pytest.mark.parametrize("alpha", [0.5, [0.5], [0.5, 0.9]])
+def test_p_values_different_alpha(alpha: Union[float, NDArray]) -> None:
+    """Test type for different alpha for p_values"""
+    result = compute_hoefdding_bentkus_p_value(r_hat, n, alpha)
+    assert isinstance(result, np.ndarray)
+
+
+@pytest.mark.parametrize("delta", [0.1, 0.2])
+def test_ltt_different_delta(delta: float) -> None:
+    """Test _ltt_procedure for different delta"""
+    assert _ltt_procedure(r_hat, alpha, delta, n)
+
+
+def test_find_lambda_control_star() -> None:
+    """Test _find_lambda_control_star"""
+    assert _find_lambda_control_star(r_hat, valid_index, lambdas)
+
+
+@pytest.mark.parametrize("delta", [0.1, 0.8])
+@pytest.mark.parametrize("alpha", [[0.5], [0.6, 0.8]])
+def test_ltt_type_output_alpha_delta(
+    alpha: NDArray,
+    delta: float
+) -> None:
+    """Test type output _ltt_procedure"""
+    valid_index, p_values = _ltt_procedure(r_hat, alpha, delta, n)
+    assert isinstance(valid_index, list)
+    assert isinstance(p_values, np.ndarray)
+
+
+@pytest.mark.parametrize("valid_index", [[[0, 1]]])
+def test_find_lambda_control_star_output(valid_index: List[List[int]]) -> None:
+    """Test _find_lambda_control_star with a list of list"""
+    assert _find_lambda_control_star(r_hat, valid_index, lambdas)
+
+
+def test_warning_valid_index_empty() -> None:
+    """Test warning sent when empty list"""
+    valid_index = [[]]  # type: List[List[int]]
+    with pytest.warns(UserWarning,
+                      match=r".*At least one sequence is empty*"):
+        _find_lambda_control_star(r_hat, valid_index, lambdas)
+
+
+def test_invalid_alpha_hb() -> None:
+    """Test error message when invalid alpha"""
+    with pytest.raises(ValueError, match=r".*Invalid alpha"):
+        compute_hoefdding_bentkus_p_value(r_hat, n, wrong_alpha)
+
+
+def test_invalid_shape_alpha_hb() -> None:
+    """Test error message when invalid alpha shape"""
+    with pytest.raises(ValueError, match=r".*Invalid alpha"):
+        compute_hoefdding_bentkus_p_value(r_hat, n, wrong_alpha_shape)
+
+
+def test_delta_none_ltt() -> None:
+    """Test error message when invalid delta"""
+    with pytest.raises(ValueError, match=r".*Invalid delta"):
+        _ltt_procedure(r_hat, alpha, wrong_delta, n)
```

Comment:
```suggestion
```suggestion
@pytest.mark.parametrize("delta", [None])
def test_delta_none_ltt(delta: Optional[float]) -> None:
    """Test error message when invalid delta"""
    with pytest.raises(ValueError, match=r".*Invalid delta"):
        _ltt_procedure(r_hat, alpha, delta, n)
```

Reply:
Done